src/components/MessageList/utils.ts

```diff
@@ -303,24 +304,29 @@ export const getGroupStyles = <
     previousMessage.customType === CUSTOM_MESSAGE_TYPE.intro ||
     previousMessage.customType === CUSTOM_MESSAGE_TYPE.date ||
     previousMessage.type === 'system' ||
+    previousMessage.type === 'error' ||
     previousMessage.attachments?.length !== 0 ||
     message.user?.id !== previousMessage.user?.id ||
-    previousMessage.type === 'error' ||
     previousMessage.deleted_at ||
     (message.reaction_groups && Object.keys(message.reaction_groups).length > 0) ||
     isMessageEdited(previousMessage);
 
   const isBottomMessage =
     !nextMessage ||
+    nextMessage.customType === CUSTOM_MESSAGE_TYPE.intro ||
     nextMessage.customType === CUSTOM_MESSAGE_TYPE.date ||
     nextMessage.type === 'system' ||
-    nextMessage.customType === CUSTOM_MESSAGE_TYPE.intro ||
+    nextMessage.type === 'error' ||
     nextMessage.attachments?.length !== 0 ||
     message.user?.id !== nextMessage.user?.id ||
-    nextMessage.type === 'error' ||
     nextMessage.deleted_at ||
     (nextMessage.reaction_groups && Object.keys(nextMessage.reaction_groups).length > 0) ||
-    isMessageEdited(message);
+    isMessageEdited(message) ||
+    (maxTimeBetweenGroupedMessages !== undefined &&
+      nextMessage.created_at &&
+      message.created_at &&
+      new Date(nextMessage.created_at).getTime() - new Date(message.created_at).getTime() >
+        maxTimeBetweenGroupedMessages);
```

Comment:
Shall we have a similar check above for the `isTopMessage`, but with `message` and `previousMessage`?

Reply:
Yes we should so that the next image is associated with top class. Let me check

Reply:
Adjusted: https://github.com/GetStream/stream-chat-react/pull/2439/commits/bd824a5223c53c3c7e0f753bea42a45287d9be39